### PR TITLE
feat(dataflow): Add SASL mechanism support in dataflow engine

### PR DESCRIPTION
**Pull request by agrski**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: scheduler/data-flow/src/main/kotlin/io/seldon/dataflow/Cli.kt [hunk 1]
```diff
@@ -34,31 +35,28 @@ object Cli {
...
+    val tlsCACertPath = Key("kafka.tls.client.ca.path", stringType)
+    val tlsKeyPath = Key("kafka.tls.client.key.path", stringType)
+    val tlsCertPath = Key("kafka.tls.client.cert.path", stringType)
+    val brokerCACertPath = Key("kafka.tls.broker.ca.path", stringType)
+    val clientSecret = Key("kafka.tls.client.secret", stringType)
+    val brokerSecret = Key("kafka.tls.broker.secret", stringType)
+    val endpointIdentificationAlgorithm = Key("kafka.tls.endpoint.identification.algorithm", stringType)
+
+    // Kafka SASL
+    val saslUsername = Key("kafka.sasl.username", stringType)
```
Comment (RafalSkolasinski) [inline scheduler/data-flow/src/main/kotlin/io/seldon/dataflow/Cli.kt:56]: So, as I understand from our discussions, changing here `sasl.username` to `kafka.sasl.username` also impacts what environmental variables must be set but this PR seems to be missing helm chart changes?

File: scheduler/data-flow/src/main/kotlin/io/seldon/dataflow/Cli.kt [hunk 2]
```diff
@@ -34,31 +35,28 @@ object Cli {
...
-    val brokerCACertPath = Key("tls.broker.ca.path", stringType)
-    val clientSecret = Key("tls.client.secret", stringType)
-    val brokerSecret = Key("tls.broker.secret", stringType)
-    val endpointIdentificationAlgorithm = Key("tls.endpoint.identification.algorithm", stringType)
-
-    // SASL
-    val saslUsername = Key("sasl.username", stringType)
-    val saslSecret = Key("sasl.secret", stringType)
-    val saslPasswordPath = Key("sasl.password.path", stringType)
+    val kafkaMaxMessageSizeBytes = Key("kafka.max.message.size.bytes", longType)
```
Comment (agrski) [inline scheduler/data-flow/src/main/kotlin/io/seldon/dataflow/Cli.kt:44]: :wrench: Add Helm value for this new parameter (needed for e.g. Confluent Cloud Kafka)

File: scheduler/data-flow/src/main/kotlin/io/seldon/dataflow/kafka/security/Sasl.kt [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+data class SaslConfig(
+    val mechanism: String,
+    val username: String,
+    val secret: String,
+    val passwordPath: FilePath
+)
+
+enum class KafkaSaslMechanisms(val mechanism: String) {
+    PLAIN("PLAIN"),
+    SCRAM_SHA_256("SCRAM-SHA-256"),
```
Comment (agrski) [inline scheduler/data-flow/src/main/kotlin/io/seldon/dataflow/kafka/security/Sasl.kt:14]: :wrench: Test this and confirm it's parsed correctly via Konfig

File: scheduler/data-flow/src/main/kotlin/io/seldon/dataflow/kafka/security/Sasl.kt [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package io.seldon.dataflow.kafka.security
+
```
Comment (agrski) [inline scheduler/data-flow/src/main/kotlin/io/seldon/dataflow/kafka/security/Sasl.kt:2]: :wrench: Re-run licence generation

File: k8s/kustomize/helm-components/patch_dataflow.yaml [hunk 5]
```diff
@@ -17,27 +17,31 @@ spec:
...
+        - name: SELDON_KAFKA_SASL_USERNAME
           value: '{{ .Values.security.kafka.sasl.client.username }}'
-        - name: SELDON_SASL_SECRET
+        - name: SELDON_KAFKA_SASL_SECRET
           value: '{{ .Values.security.kafka.sasl.client.secret }}'
-        - name: SELDON_SASL_PASSWORD_PATH
+        - name: SELDON_KAFKA_SASL_PASSWORD_PATH
           value: '{{ .Values.security.kafka.sasl.client.passwordPath }}'
-        - name: SELDON_TLS_BROKER_SECRET
+        - name: SELDON_KAFKA_TLS_BROKER_SECRET
```
Comment (RafalSkolasinski) [inline k8s/kustomize/helm-components/patch_dataflow.yaml:44]: Duplicated key with above

File: k8s/kustomize/helm-components/patch_dataflow.yaml [hunk 6]
```diff
@@ -17,27 +17,31 @@ spec:
...
+        - name: SELDON_KAFKA_TLS_CLIENT_CERT_PATH
           value: '{{ .Values.security.kafka.ssl.client.crtPath }}'
-        - name: SELDON_TLS_CLIENT_CA_PATH
+        - name: SELDON_KAFKA_TLS_CLIENT_CA_PATH
           value: '{{ .Values.security.kafka.ssl.client.caPath }}'
-        - name: SELDON_SASL_MECHANISM
+        - name: SELDON_KAFKA_TLS_BROKER_SECRET
+          value: '{{ .Values.security.kafka.ssl.client.brokerValidationSecret }}'
+        - name: SELDON_KAFKA_TLS_BROKER_CA_PATH
+          value: '{{ .Values.security.kafka.ssl.client.brokerCa }}'
```
Comment (RafalSkolasinski) [inline k8s/kustomize/helm-components/patch_dataflow.yaml:33]: Should be `.Values.security.kafka.ssl.client.brokerCaPath` so rebase did not happen well?

**Comment by RafalSkolasinski** [comment]
Image does not build 
```
 ---> Running in c19244ba07dc

(microdnf:1): librhsm-WARNING **: 12:22:09.201: Found 0 entitlement certificates

(microdnf:1): librhsm-WARNING **: 12:22:09.202: Found 0 entitlement certificates
error: No installed package matches 'librpmio'
The command '/bin/sh -c microdnf remove -y microdnf libdnf rpm-libs rpm  libsolv librpmio libmodulemd curl-minimal' returned a non-zero code: 1
make: *** [Makefile:158: docker-build-dataflow] Error
```

**Comment by RafalSkolasinski** [comment]
fixed in 33e57d1d01c36e7ce6edb603eb2762f95d7ae6f8

File: k8s/kustomize/helm-components/patch_dataflow.yaml (hunk 6 above)
Comment (RafalSkolasinski) [reply k8s/kustomize/helm-components/patch_dataflow.yaml:33]: Fixed the kustomize and chart issue via https://github.com/RafalSkolasinski/seldon-core/commit/3ab189dd80fb5f5e96c5ed49bad6fbd7afc9521f but did not want to push into your fork.

**Comment by RafalSkolasinski** [comment]
Build image and tried to deploy to k8s but no luck. 
```
time="2023-05-10T12:44:49Z" level=info msg="Setting log level to debug"
time="2023-05-10T12:44:49Z" level=info msg="Setting namespace from k8s file to seldon"
time="2023-05-10T12:44:49Z" level=info msg="No tracing path provided so setting NOOP TraceProvider" func=loadConfig source=TraceProvider
time="2023-05-10T12:44:49Z" level=fatal msg="Failed to load Kafka config" error="open /mnt/config/kafka.json: no such file or directory"
```

**Comment by RafalSkolasinski** [comment]
Was getting of course error about certificates not being found in specified locaton so added them like in Pipeline / Model Gateway images: https://github.com/RafalSkolasinski/seldon-core/commit/6c6522539f09956bf840951fc05e465e1f279d00

Now I am getting
```
Exception in thread "main" org.apache.kafka.common.config.ConfigException: Invalid value 1000000000 for configuration receive.buffer.bytes: Expected value to be a 32-bit integer, but it was a java.lang.Long 
```

**Pull request merged**